lib/ts/signals.cc

```diff
@@ -46,7 +46,7 @@ signal_check_handler(int signal, signal_handler_t handler)
     sigact = (void *)oact.sa_sigaction;
   }
 
-  if (sigact != handler) {
+  if (sigact != (void *)handler) {
```

Comment:
Actually, I'm not gonna change this, because everything else (in a million places) uses the C-style cast already.